redfish-core/lib/assembly.hpp

```diff
@@ -321,6 +368,80 @@ inline void setAssemblylocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
+        {
+            if (!json_util::readJson(req, asyncResp->res, "Assemblies", Assemblies, "Oem", oem))
+            {
+                return;
+            }
+            if (oem)
+            {
+                std::optional<nlohmann::json> openbmc;
+                if (!json_util::readJson(*oem, asyncResp->res, "OpenBMC", openbmc))
+                {
+                    BMCWEB_LOG_ERROR << "Property Missing ";
+                    messages::propertyMissing(asyncResp->res, "OpenBMC");
+                    return;
+                }
+                if (openbmc)
+                {
+                    std::optional<bool> readytoremove;
+                    if (!json_util::readJson(*openbmc, asyncResp->res,
+                                             "ReadyToRemove", readytoremove))
+                    {
+                        BMCWEB_LOG_ERROR << "Property Missing ";
+                        messages::propertyMissing(asyncResp->res, "ReadyToRemove");
```

Comment:
Just so you know I don't believe this actually returns false when the property is missing 

I think you need a 


        if (!readytoremove)
        {
                    messages::propertyMissing(asyncResp->res,
                                      pathString + "/ReadyToRemove");
        }

Reply:
And this error here should be like 

                messages::propertyValueFormatError(asyncResp->res, *gateway,
                                                   pathString + "/ReadyToRemove");
                                                   
                                                   Have a look at https://github.com/openbmc/bmcweb/blob/e05aec50f10116e6dda7e377bc61799aa5b7c166/redfish-core/lib/hypervisor_system.hpp#L647

Reply:
Should we not code this similar to the other OEM cases?
See: https://github.com/ibm-openbmc/bmcweb/blob/f3aa8d46eeaf72a517997f029dc42e0b60053334/redfish-core/lib/account_service.hpp#L2212

Reply:
Tested above for return of true or false.
The value of "(!json_util::readJson(*openbmc, asyncResp->res, "ReadyToRemove", readytoremove))"
evaluates to "1" when the property is missing.
```
                 if (openbmc)
                 {
                     std::optional<bool> readytoremove;

                     //TEST VARIABLE ADDED
                     auto READY2REMOVE_VALUE
                           = (!json_util::readJson(*openbmc, asyncResp->res, "ReadyToRemove", readytoremove));
                     std::cout<<"HERE IS VALUE OF READY2REMOVE_VALUE:"<< READY2REMOVE_VALUE <<std::endl;

                     if (!json_util::readJson(*openbmc, asyncResp->res,
                                              "ReadyToRemove", readytoremove))
                     {
                         BMCWEB_LOG_ERROR << "Property Missing ";
                         messages::propertyMissing(asyncResp->res, "ReadyToRemove");
                         return;
                     }
```
root@p10bmc:/tmp# HERE IS VALUE OF READY2REMOVE_VALUE:1
(2022-02-24 04:56:51) [ERROR "assembly.hpp":401] Property Missing